### ISSUE TITLE
The mso_user resources is not supported on ND-based MSO/NDO. Use ND provider for manipulating users on ND-based MSO/NDO.

### ISSUE THREAD

**Issue by soumukhe** [CiscoDevNet/terraform-provider-mso#79]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
aciadmin@ubuntu-jump:~/Terraform/AWS-EKS-Terraform/AWS_Ter_EKSCTL/importUserID$ terraform version
Terraform v1.0.11
on linux_amd64
+ provider registry.terraform.io/ciscodevnet/mso v0.3.2
```
### MSO version

* 3.6.0.88 running on vND 2.1.1e

### APIC version and APIC Platform for Site Level Resources

* 25.0(1.57)  AWS and Azure

### Affected Resource(s)

* NDO running on ND (3.6.0.88 in this case, but this is not tied to this release)

### Terraform Configuration Files

```hcl
aciadmin@ubuntu-jump:~/Terraform/AWS-EKS-Terraform/AWS_Ter_EKSCTL/importUserID$ cat import.tf 

terraform {
  required_providers {
    mso = {
      source = "ciscodevnet/mso"
    }
  }
  required_version = ">= 0.13"
}

provider "mso" {
  username = var.creds.username
  password = var.creds.password
  url      = var.creds.url
  domain   = var.creds.domain.  # domain is not associated, but that's a different issue and "MSO provider v0.3.2 is ignoring domain config: #74" has been opened for that
  insecure = "true"
  platform = "nd"
}


resource "mso_user" "user1" {
  username = "admin"
}

#usage: terraform import mso_user.user1 {user_id}
```

### Results:
```
aciadmin@ubuntu-jump:~/Terraform/AWS-EKS-Terraform/AWS_Ter_EKSCTL/importUserID$ terraform import mso_user.user1 {user_id}
mso_user.user1: Importing from ID "{user_id}"...
╷
│ Error: The mso_user resources is not supported on ND-based MSO/NDO. Use ND provider for manipulating users on ND-based MSO/NDO.
│ 
│ 
╵
```

### Comments:

This is a problem, because we cannot easily associate user to Tenant without knowing the userID.
The only way to do this is to associate the user to the tenant manually and then do a api call at:  https://{msoIP}/mso/api/v1/tenants and extract the userID from there

**Comment by lhercot** [comment]
You should use a data source not a ressource to query the user ID. Can you test with a data source and let me know if there is an issue with it?

**Comment by soumukhe** [comment]
Thank You Lionel.  Yes, I had tried that also, but get a blank output.  I think the problem is because users are defined in the base ND platform and not NDO.  

# Please see below (soumukhe is remote user):

```ccl
data "mso_user" "users" {
  username = "soumukhe"
  }

output "user_info" {
  value = data.mso_user.users
}
```
# Results (in this case soumukhe is a remote user):

```json
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

user_info = {
  "account_status" = tostring(null)
  "domain" = tostring(null)
  "email" = tostring(null)
  "first_name" = tostring(null)
  "id" = "{}"
  "last_name" = tostring(null)
  "phone" = tostring(null)
  "roles" = toset([
    {
      "access_type" = "WritePriv"
      "roleid" = "admin"
    },
  ])
  "user_password" = "{}"
  "username" = "soumukhe"
}
```

# Also tried with local user "admin"
```hcl
data "mso_user" "users" {
  username = "admin"
  }

output "user_info" {
  value = data.mso_user.users
}
```
# Results with local user admin:

```json
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

user_info = {
  "account_status" = "Active"
  "domain" = tostring(null)
  "email" = tostring(null)
  "first_name" = "admin"
  "id" = "{}"
  "last_name" = tostring(null)
  "phone" = tostring(null)
  "roles" = toset([
    {
      "access_type" = "WritePriv"
      "roleid" = "admin"
    },
  ])
  "user_password" = "{}"
  "username" = "admin"
}
```

**Comment by lhercot** [comment]
This issue should be fixed in v0.3.3. Thanks for reporting it.
